src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+
+    return lines;
+  }
+
+  public async load(fileName: string, keep_current_data = true): Promise<boolean> {
+    if (!keep_current_data) {
+      this.sections = [];
+    }
+
+    const ins = fs.createReadStream(fileName);
+    ins.on("error", (e) => {
+      console.log("failed to read file with error", e);
+      alert(e);
+    });
+    const rl = readline.createInterface({
+      input: ins,
+      terminal: false,
+    });
+    let current_section = undefined;
+    let first_line = true;
+    for await (let line of rl) {
+      //console.log(line);
+      // Skips the UTF-8 BOM at the start of files. Notepad likes to add this.
+      if (first_line && line.substr(0, 3) === "\xEF\xBB\xBF") {
+        line = line.slice(3);
+      }
+      first_line = false;
+
+      //section line
+      if (line[0] === "[") {
+        //console.log(line, line[0]);
+        const endpos = line.indexOf("]");
+        if (endpos !== -1) {
+          //we have a new section
+          const sub = line.substr(1, endpos - 1);
+          //console.log(sub);
+          current_section = this.getOrCreateSection(sub);
+          //console.log(current_section);
+        }
+      } else {
+        if (current_section !== undefined) {
+          const [key, value] = this.parseLine(line);
+
+          // Lines starting with '$', '*' or '+' are kept verbatim.
+          // Kind of a hack, but the support for raw lines inside an
+          // INI is a hack anyway.
+          if (
+            (key === null && value === null) ||
+            (line.length !== 0 && ["$", "+", "*"].some((val) => line[0] === val))
+          ) {
+            current_section.m_lines.push(line);
+          } else if (key !== null && value !== null) {
+            current_section.set(key, value);
+          }
+        }
+      }
+    }
+
+    return true;
+  }
+
+  public save(filePath: string): boolean {
+    const out = fs.createWriteStream(filePath);
+
+    out.on("error", (e) => {
+      console.log("failed to write file with error", e);
+      alert(e);
+    });
+
+    this.sections.forEach((section) => {
+      // originally section.name was only written if the section was non-empty,
+      // but that goes against us wanting to always show the Gecko section
+      out.write(`[${section.name}]\n`);
+
+      if (section.keys_order.length === 0) {
+        section.m_lines.forEach((line) => {
+          out.write(`${line}\n`);
+        });
+        out.write("\n");
+      } else {
+        section.keys_order.forEach((kvit) => {
+          const value = section.values.get(kvit);
+          out.write(`${kvit}=${value}\n`);
+        });
+        out.write("\n");
+      }
+    });
+
+    out.end();
+    out.close();
+
+    return true;
+  }
+}
+
+/**
+ * The Section class
+ */
+export class Section {
+  public name: string;
+  public keys_order: string[];
+  public m_lines: string[];
+  public values: Map<string, string>;
+
+  public constructor(_name: string) {
```

Comment:
```suggestion
  public constructor(name: string) {
```

Similar to the other suggestion, we probably wanna stick with the conventions laid out in this repo, where we haven't really been using underscore prefixes for params.